File: python_solution/vertex.py
class Vertex:
	def __init__(self, current_bunnies = 0, mec = 0):
		self.in_edges = []
		self.out_edges = []
		self.current_bunnies = current_bunnies
		self.mec = mec
		self.used_out_edge_count = 0
		self.used_in_edge_count = 0
		self.needed_bunnies = 0
		self.sum_ieew = 0

	def ins_equal(self):
		return self.used_in_edge_count == len(self.in_edges)

	def outs_equal(self):
		return self.used_out_edge_count == len(self.out_edges)

	def update_needed_bunnies(self):
		self.needed_bunnies = self.mec - self.sum_ieew

File: vertex.rb
class Vertex
	attr_accessor :in_edges,
		:out_edges,
		:current_bunnies,
		:mec,
		:used_out_edge_count,
		:used_in_edge_count,
		:needed_bunnies,
		:sum_ieew

	def initialize(current_bunnies = 0, mec = 0)
		@in_edges = []
		@out_edges = []
		@current_bunnies = current_bunnies
		@mec = mec
		@used_out_edge_count = 0
		@used_in_edge_count = 0
		@needed_bunnies = 0
		@sum_ieew = 0
	end

	def ins_equal?
		@used_in_edge_count == @in_edges.length
	end

	def outs_equal?
		@used_out_edge_count == @out_edges.length
	end

	def update_needed_bunnies!
		@needed_bunnies = @mec - @sum_ieew
	end
end
